src/main/java/net/snowflake/client/core/SessionUtilExternalBrowser.java

```diff
@@ -265,6 +269,7 @@ private String generateProofKey() {
   void authenticate() throws SFException, SnowflakeSQLException {
     ServerSocket ssocket = this.getServerSocket();
     try {
+      ssocket.setSoTimeout(getBrowserResponseTimeout());
```

Comment:
I tested your PR and it works as expected but let's improve the error message - on timeout we receive exception `java.net.SocketTimeoutException: Accept timed out`

please catch it explicitly and add message that `External browser authentication failed within timeout of {} seconds`